compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -428,8 +428,18 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                 ty => bug!("unexpected DefiningTy {:?}", ty),
             };
 
-            let upvar_def_span = self.infcx.tcx.hir().span(upvar);
-            let upvar_span = self.infcx.tcx.upvars_mentioned(def_id).unwrap()[&upvar].span;
+            let captured_place = &self.upvars[upvar.index()].place;
+            let defined_hir = match captured_place.place.base {
+                rustc_middle::hir::place::PlaceBase::Local(hirid) => hirid,
+                rustc_middle::hir::place::PlaceBase::Upvar(up) => up.var_path.hir_id,
+                _ => panic!(
```

Comment:
What should we do here?

Reply:
I would be inclined to make `defined_hir` an Option, and just skip the `span_label` call in this case.